File: scripts/check-erol-ui.ts
import pg from "pg";

const { Pool } = pg;

const pool = new Pool({
  connectionString:
    "postgresql://postgres:518518Erkan@localhost:5432/optimusvet",
});

async function checkErolUI() {
  const client = await pool.connect();

  try {
    console.log("🔍 EROL DEMİR - UI GÖRÜNÜM KONTROLÜ\n");
    console.log("=".repeat(80));

    // Müşteriyi bul
    const customerResult = await client.query(`
      SELECT * FROM customers WHERE code = 'MUS-173'
    `);

    const customer = customerResult.rows[0];

    console.log(`\n👤 MÜŞTERİ: ${customer.name}`);
    console.log(
      `   Bakiye: ₺${parseFloat(customer.balance).toLocaleString("tr-TR", { minimumFractionDigits: 2 })}\n`,
    );

    // Satışları say
    const salesResult = await client.query(
      `
      SELECT COUNT(*) as count, SUM(total) as total
      FROM transactions
      WHERE "customerId" = $1 AND type = 'SALE'
    `,
      [customer.id],
    );

    const salesData = salesResult.rows[0];
    console.log(`🛒 SATIŞLAR:`);
    console.log(`   Sayı: ${salesData.count}`);
    console.log(
      `   Toplam: ₺${parseFloat(salesData.total || 0).toLocaleString("tr-TR", { minimumFractionDigits: 2 })}\n`,
    );

    // Ödemeleri say
    const paymentsResult = await client.query(
      `
      SELECT COUNT(*) as count, SUM(total) as total
      FROM transactions
      WHERE "customerId" = $1 AND type = 'CUSTOMER_PAYMENT'
    `,
      [customer.id],
    );

    const paymentsData = paymentsResult.rows[0];
    console.log(`💰 ÖDEMELER:`);
    console.log(`   Sayı: ${paymentsData.count}`);
    console.log(
      `   Toplam: ₺${parseFloat(paymentsData.total || 0).toLocaleString("tr-TR", { minimumFractionDigits: 2 })}\n`,
    );

    // UI'da gösterilen işlemleri kontrol et (Son işlemler tablosu)
    console.log("=".repeat(80));
    console.log("📋 UI'DA GÖSTERILEN İŞLEMLER (Son İşlemler Tablosu):\n");

    const uiTransactions = await client.query(
      `
      SELECT 
        t.id,
        t.type,
        t.date,
        t.total,
        t.status,
        c.name as customer_name,
        a.name as animal_name
      FROM transactions t
      LEFT JOIN customers c ON c.id = t."customerId"
      LEFT JOIN animals a ON a.id = t."animalId"
      WHERE t."customerId" = $1
      ORDER BY t.date DESC
      LIMIT 20
    `,
      [customer.id],
    );

    console.log(`Toplam ${uiTransactions.rows.length} işlem bulundu:\n`);

    uiTransactions.rows.forEach((tx, idx) => {
      const typeLabels: Record<string, string> = {
        SALE: "🛒 Satış",
        CUSTOMER_PAYMENT: "💰 Ödeme",
        TREATMENT: "🏥 Tedavi",
      };

      const typeLabel = typeLabels[tx.type] || tx.type;
      const amount = parseFloat(tx.total);
      const date = new Date(tx.date);

      console.log(`${idx + 1}. ${typeLabel}`);
      console.log(`   Tarih: ${date.toLocaleDateString("tr-TR")}`);
      console.log(
        `   Tutar: ₺${amount.toLocaleString("tr-TR", { minimumFractionDigits: 2 })}`,
      );
      console.log(`   Durum: ${tx.status}`);
      if (tx.animal_name) {
        console.log(`   Hayvan: ${tx.animal_name}`);
      }
      console.log("");
    });

    // Satış ve ödeme dağılımı
    console.log("=".repeat(80));
    console.log("📊 İŞLEM DAĞILIMI:\n");

    const typeDistribution = await client.query(
      `
      SELECT 
        type,
        COUNT(*) as count,
        SUM(total) as total
      FROM transactions
      WHERE "customerId" = $1
      GROUP BY type
      ORDER BY count DESC
    `,
      [customer.id],
    );

    typeDistribution.rows.forEach((row) => {
      const typeLabels: Record<string, string> = {
        SALE: "Satış",
        CUSTOMER_PAYMENT: "Ödeme",
        TREATMENT: "Tedavi",
      };

      console.log(`${typeLabels[row.type] || row.type}:`);
      console.log(`  Sayı: ${row.count}`);
      console.log(
        `  Toplam: ₺${parseFloat(row.total).toLocaleString("tr-TR", { minimumFractionDigits: 2 })}\n`,
      );
    });

    // SORUN TESPİTİ
    console.log("=".repeat(80));
    console.log("🔍 SORUN ANALİZİ:\n");

    if (parseInt(salesData.count) === 0) {
      console.log("❌ SORUN: Satış kaydı yok!");
      console.log("   Ama ödeme var. Bu normal değil.\n");
    } else if (parseInt(salesData.count) < parseInt(paymentsData.count)) {
      console.log("⚠️  UYARI: Ödeme sayısı satış sayısından fazla!");
      console.log(
        `   Satış: ${salesData.count} | Ödeme: ${paymentsData.count}\n`,
      );
    } else {
      console.log("✅ Satış ve ödeme kayıtları dengeli görünüyor.\n");
    }

    // UI filtreleme kontrolü
    console.log("💡 UI KONTROL ÖNERİSİ:");
    console.log('   Müşteri detay sayfasındaki "Son İşlemler" tablosunda:');
    console.log(
      "   - Tüm işlem tipleri gösteriliyor mu? (SALE, CUSTOMER_PAYMENT, TREATMENT)",
    );
    console.log("   - Tarih filtresi var mı?");
    console.log("   - Pagination doğru çalışıyor mu?\n");

    console.log("=".repeat(80));
  } catch (error) {
    console.error("❌ HATA:", error);
  } finally {
    client.release();
    await pool.end();
  }
}

checkErolUI();

File: scripts/check-erol-ui.py
import psycopg2

# Database connection
conn = psycopg2.connect(
    host="localhost",
    port=5432,
    database="optimusvet",
    user="postgres",
    password="518518Erkan"
)

cur = conn.cursor()

print("=== TURKAY DEMIRHAN (MUS-063) KONTROL ===\n")

# Müşteriyi kod ile bul
cur.execute("""
    SELECT id, name, code, balance
    FROM customers
    WHERE code = 'MUS-063'
""")

customer = cur.fetchone()

if not customer:
    print("❌ Müşteri bulunamadı!")
    cur.close()
    conn.close()
    exit()

customer_id, customer_name, customer_code, customer_balance = customer

print(f"Müşteri: {customer_name}")
print(f"Kod: {customer_code}")
print(f"Güncel Bakiye: {float(customer_balance):,.2f} TL\n")

# Tüm satışlar
cur.execute("""
    SELECT COUNT(*), SUM(total)
    FROM transactions
    WHERE "customerId" = %s
    AND type = 'SALE'
""", (customer_id,))

sales_count, sales_total = cur.fetchone()

print(f"=== SATIŞLAR ===")
print(f"Toplam: {sales_count} adet")
print(f"Tutar: {float(sales_total or 0):,.2f} TL\n")

# Tüm tahsilatlar (CUSTOMER_PAYMENT type'ı)
cur.execute("""
    SELECT COUNT(*), SUM(total)
    FROM transactions
    WHERE "customerId" = %s
    AND type = 'CUSTOMER_PAYMENT'
""", (customer_id,))

payments_count, payments_total = cur.fetchone()

print(f"=== TAHSİLATLAR ===")
print(f"Toplam: {payments_count} adet")
print(f"Tutar: {float(payments_total or 0):,.2f} TL\n")

# Hesaplanan bakiye
calculated_balance = float(sales_total or 0) - float(payments_total or 0)

print(f"=== HESAP ÖZETİ ===")
print(f"Satış Toplamı: {float(sales_total or 0):,.2f} TL")
print(f"Tahsilat Toplamı: {float(payments_total or 0):,.2f} TL")
print(f"Hesaplanan Bakiye: {calculated_balance:,.2f} TL")
print(f"Database Bakiye: {float(customer_balance):,.2f} TL")

if abs(calculated_balance - float(customer_balance)) < 0.01:
    print("✅ Bakiye doğru!")
else:
    print(f"⚠️ Bakiye farkı: {abs(calculated_balance - float(customer_balance)):,.2f} TL")

print("\n=== ESKİ SİSTEM KARŞILAŞTIRMASI ===")
print("Eski Sistem:")
print("  Toplam Borç: 68.003,00 TL (94 satış)")
print("  Toplam Ödeme: 58.529,00 TL (40 tahsilat)")
print("  Bakiye: 9.474,00 TL")
print("\nYeni Sistem:")
print(f"  Toplam Satış: {float(sales_total or 0):,.2f} TL ({sales_count} satış)")
print(f"  Toplam Tahsilat: {float(payments_total or 0):,.2f} TL ({payments_count} tahsilat)")
print(f"  Bakiye: {calculated_balance:,.2f} TL")

sales_diff = float(sales_total or 0) - 68003
payments_diff = float(payments_total or 0) - 58529
balance_diff = calculated_balance - 9474

print("\nFarklar:")
print(f"  Satış Farkı: {sales_diff:,.2f} TL")
print(f"  Tahsilat Farkı: {payments_diff:,.2f} TL")
print(f"  Bakiye Farkı: {balance_diff:,.2f} TL")

# Durum dağılımı
cur.execute("""
    SELECT status, COUNT(*), SUM(total)
    FROM transactions
    WHERE "customerId" = %s
    AND type = 'SALE'
    GROUP BY status
    ORDER BY status
""", (customer_id,))

status_breakdown = cur.fetchall()

print("\n=== DURUM DAĞILIMI ===")
for status, count, total in status_breakdown:
    print(f"{status}: {count} adet - {float(total):,.2f} TL")

print("\n=== SONUÇ ===")
if abs(balance_diff) < 100:
    print("✅ Veriler eski sistemle uyumlu!")
else:
    print("⚠️ Yeni sistemde daha fazla kayıt var!")
    print(f"   Eski sistem: 94 satış, 40 tahsilat")
    print(f"   Yeni sistem: {sales_count} satış, {payments_count} tahsilat")
    print(f"   Fark: {sales_count - 94} satış, {payments_count - 40} tahsilat")

cur.close()
conn.close()
